fee_allocator/accounting/distribution.py

```diff
@@ -119,21 +118,43 @@ def re_distribute_incentives(
-        # Distribute  1/6th of incentives_taken_from_aura_market to each of the top 6 pools, while subtracting the same amount from bal_incentives
-        for pool_id, _data in top_6_pools:
-            if _data['total_incentives'] > 0:
-                # TODO:  Still an edge case here.  If amount per pool is not allocated, we'll still underallocate to
-                #   aura.  It shouldn't happen on the 6 largest pools system wide though.
-                incentives[pool_id]['aura_incentives'] += min(amount_per_pool, _data['bal_incentives'])
-                incentives[pool_id]['bal_incentives'] -= min(amount_per_pool, _data['bal_incentives'])
+        debt_repaid = 0
+        ## Find how many pools ever could be over min_aura_incentive
+        pools_over_aura_min = [pool_id for pool_id, _data in incentives.items() if
+                               _data['aura_incentives'] >= min_aura_incentive]
+        num_pools_over_min = len(pools_over_aura_min)
+        ## Figure out how much to shift per pool using an even split
+        if num_pools_over_min == 0:
+            print(
+                f"WARNING: {incentives[pool_id]['chain']}:{pool_id} has no pools over min_aura_incentive, but owes {debt_to_aura_market} to the aura market.  Debt will not be repaid.")
+            amount_per_pool = 0
+        else:
+            amount_per_pool = round(debt_to_aura_market / num_pools_over_min, 4)
+        ## Check if all ov our
+        for pool_id in pools_over_aura_min:
+            ## TODO: Consider this logic as an additional test/more sensitive handlingthat could allow pool selection based
+            #   on total_incentives instead of aura incentives
+            #   if (incentives['aura_incentives'] + amount_per_pool) < min_aura_incentive:
+            #         num_pools_over_min -= 1
+            # Distribute the aura_debt to the pools that are over the min_aura_incentive
+            if incentives[pool_id]['total_incentives'] > 0:
+                # TODO:  Need to think about edge cases here and watch them.
+                incentives[pool_id]['aura_incentives'] += min(amount_per_pool, incentives[pool_id]['bal_incentives'])
+                incentives[pool_id]['bal_incentives'] -= min(amount_per_pool, incentives[pool_id]['bal_incentives'])
+                debt_repaid += min(amount_per_pool, incentives[pool_id]['bal_incentives'])
+            if debt_to_aura_market - debt_repaid > 0:
+                print(
+                    f"{incentives[pool_id]['chain']}:{pool_id}  remaining debt to aura market: {debt_to_aura_market}, Debt repaid: {debt_repaid}, debt remaining: {debt_to_aura_market - debt_repaid}")
     return incentives
 
-def add_last_join_exit(incentives: Dict[str, Dict], chain: Chains, alertTimeStamp: Optional[int] = None) -> Dict[str, Dict]:
+
+def add_last_join_exit(incentives: Dict[str, Dict], chain: Chains, alertTimeStamp: Optional[int] = None) -> Dict[
+    str, Dict]:
```

Comment:
this looks off

Reply:
time for black ci linter in this repo hehe

Reply:
I'm still trying to figure out how to get it to work in my IDE.  I thought I was using black, but this is what it did.  Feel free to setup the linkter.  Don't think we should block this though/that's another pr.